### ISSUE TITLE
[Bug WSL] bun install blocking at the end

### ISSUE THREAD

**Issue by MidKnightXI** [oven-sh/bun#481]
So I just installed bun on my WSL and wanted to try it on an existing project I was working on.

I tried to install the deps via `bun install` and all worked perfectly (it was slower than yarn or npm but we are not here for this) and at the end of the install bun blocked on the installation of the last package.

I `^C` then retried and everything was installed so I just wanted to report that bug.

package.json:

```json
{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "author": "MidKnightXI",
  "license": "MIT",
  "type": "module",
  "scripts": {
    "format:check": "prettier --check .",
    "format:write": "prettier --write .",
    "lint:check": "eslint .",
    "lint:fix": "eslint --fix .",
    "bundle": "rollup --config",
    "build": "rollup -c && pkg -o test -t node16-win-x64,node16-darwin-x64 bundle.js",
    "build:win": "rollup -c && pkg -o test -t node16-win-x64 bundle.js",
    "build:win-arm": "rollup -c && pkg -o test -t node16-win-arm64 bundle.js",
    "build:darwin": "rollup -c && pkg -o test -t node16-darwin-x64 bundle.js"
  },
  "dependencies": {
    "asar": "^3.1.0",
    "child_process": "^1.0.2",
    "fs": "^0.0.1-security",
    "path": "^0.12.7",
    "rimraf": "^3.0.2"
  },
  "devDependencies": {
    "eslint": "^8.17.0",
    "eslint-config-prettier": "^8.5.0",
    "pkg": "^5.7.0",
    "prettier": "^2.6.2",
    "rollup": "^2.75.6"
  }
}

```

**Comment by ghost** [comment]
https://github.com/Jarred-Sumner/bun#bun-install-is-stuck

**Comment by MidKnightXI** [comment]
> https://github.com/Jarred-Sumner/bun#bun-install-is-stuck

I, unfortunately, can't reproduce the bug anymore ...

**Comment by FrameMuse** [comment]
I have the same issue here are my logs. I'm dropping two log files, first is a dry bun install and the second is the following retry attempt (which is successful).

[logs-1-failure.txt](https://github.com/oven-sh/bun/files/9275060/logs-1-failure.txt)
[logs-1-retry-success.txt](https://github.com/oven-sh/bun/files/9275062/logs-1-retry-success.txt)

I also dropped it in a discord server in @help chat.

### Details

Bug exists when it's a first-time install (when there is no `.bunlock` yet) and then after it appears, installing again with existing `.bunlock` will result in success.

**Comment by Jarred-Sumner** [comment]
Should be fixed as of Bun v0.1.12. Please comment if it isn't.

**Comment by FrameMuse** [comment]
Yes, this seems to be solved (for version `0.2.0`)
